### PR TITLE
[pvr] - expose timeshift info to GUI

### PR DESCRIPTION
**Pull request by FernetMenta**
see title

### CODE REVIEW

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by ksooo** [comment]
Yeeeeeeees! ;-)

**Comment by FernetMenta** [comment]
@hudokkow do you have time to adapt addons after this goes in?

**Comment by hudokkow** [comment]
@FernetMenta, sure.
If I'm reading it right, for minimal API support the only thing needed is
```
bool IsTimeshifting(void) { return false; }
```
and respective bumps, correct? Add-on authors can do the rest of the work later.

**Comment by FernetMenta** [comment]
@hudokkow correct, you can skip vnsi because it supports it already.

**Comment by hudokkow** [comment]
OK. I'll work on it after lunch.
btw, great work!

**Comment by hudokkow** [comment]
WIN32 failure for some reason. jenkins build this please

**Comment by MartijnKaijser** [comment]
@FernetMenta don't forget to bump https://github.com/xbmc/xbmc/blob/master/addons/xbmc.pvr/addon.xml

**Comment by FernetMenta** [comment]
thanks @MartijnKaijser, xml is updated

**Comment by hudokkow** [comment]
Apparently jenkins doesn't like this one...
jenkins build this please

**Comment by ksooo** [comment]
@FernetMenta much appreciated. What about the skin stuff? Is anybody working on adapting (at least) Confluence to support this feature?

**Comment by hudokkow** [comment]
Not that I know off but @FernetMenta already had something cooked, IIRC

**Comment by FernetMenta** [comment]
I use this one: https://github.com/FernetMenta/xbmc/commit/5c95881d5c2e5d40b2559ca19caec928f5a49bdc
but due to protests on the forum I did not submit it.

Sure this can be improved but you need to create a new gui control first.

**Comment by ksooo** [comment]
@FernetMenta thx. I've read the forum thread. Unfortunately I don't have a clue on how to create such a control. OTOH GUI controls are not my natural playground, when it comes to personal interests.

File: xbmc/pvr/PVRGUIInfo.cpp [hunk 1]
```diff
@@ -432,6 +477,11 @@ int CPVRGUIInfo::TranslateIntInfo(DWORD dwInfo) const
     else
       iReturn = 0xFF;
   }
+  else if (dwInfo == PVR_TIMESHIFT_PROGRESS)
+  {
+    iReturn = (int) ((float)(m_iTimeshiftPlayTime - m_iTimeshiftStartTime) /
```
Comment (xhaggi) [inline xbmc/pvr/PVRGUIInfo.cpp]: c++ styled casts?

**Comment by xhaggi** [comment]
great :)

**Comment by xhaggi** [comment]
@ksooo I'm not sure if we really need a new gui control. ATM xperience1080 use different progress controls for showing the player cache state, progress, seek progress etc. which looks good imo https://github.com/xhaggi/Xperience1080/blob/master/1080i/DialogSeekBar.xml#L43. could someone point me to the forum thread you are talking about?

**Comment by ksooo** [comment]
@xhaggi http://forum.kodi.tv/showthread.php?tid=226287

**Comment by xhaggi** [comment]
@ksooo thanks

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by phil65** [comment]
@FernetMenta Thx for this. Should I take a look at Confluence implementation or is someone else already doin it?

**Comment by FernetMenta** [comment]
whoever takes this is very welcome. I am not aware that anybody is working on it.

**Pull request merged**